primitives/state-machine/src/proving_backend.rs

```diff
@@ -115,7 +115,7 @@ where
 #[derive(Default)]
 struct ProofRecorderInner<Hash> {
 	/// All the records that we have stored so far.
-	records: HashMap<Hash, Option<DBValue>>,
+	records: BTreeMap<Hash, Option<DBValue>>,
```

Comment:
I did read the whole discussion, so my question may be redundant, but why not just sort the storage proof Vec in `to_storage_proof`.

Reply:
Yeah, that should also make the produced proof deterministic and requires smaller code changes, @bkchr any thoughts?

Reply:
Actually btreeset usage in storage proof does this.
thinking sorting by hash would be faster, may not be that relevant

Reply:
> Actually btreeset usage in storage proof does this. thinking sorting by hash would be faster, may not be that relevant

Exactly. There is no difference.